packages/accordion/src/index.js

```diff
@@ -0,0 +1,480 @@
+
+  // `allowMultiple` can only be false in an uncontrolled component.
+  // The same is true for `allowToggle` which can also only be false if
+  // `allowMultiple` is false.
+  const allowMultiple = isControlled || _allowMultiple;
+  const allowToggle = allowMultiple || _allowToggle;
+
+  // Even though we're calling the component uncontrolled, we'll track the
+  // state internally rather than give control to the DOM. This helps us
+  // create a disabled disclosure, which is not available for plain `details`
+  // or `summary` elements. Further, there are some inconsistencies still in
+  // how some devices and screen readers handle `details`/`summary`, so we
+  // should let the developer determine if a button toggle better fits their
+  // browser support needs.
+  const [activePanelIndices, setActivePanelIndices] = useState([]);
+  const _id = useId();
+
+  // Using state here in addition to a ref so we can move focus around
+  // when the user navigates with arrow keys
+  const [focusedItem, setFocusedItem] = useState(null);
+  const userInteractedRef = useRef(false);
+
+  warning(
+    !(isControlled && !wasControlled),
+    "Accordion is changing from controlled to uncontrolled. Accordion should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled Accordion for the lifetime of the component. Check the `index` prop being passed in."
+  );
+
+  warning(
+    !(!isControlled && wasControlled),
+    "Accordion is changing from uncontrolled to controlled. Accordion should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled Accordion for the lifetime of the component. Check the `index` prop being passed in."
+  );
+
+  warning(
+    !(isControlled && readOnly),
+    "The `readOnly` prop on Accordion was used with controlled state. This will result in buggy behavior. If the accordion does not need toggle state, you should use an uncontrolled component instead. Switch to uncontrolled or remove the `readOnly` prop."
+  );
+
+  // Loop through children and find all `disabled` items. This will allow us
+  // to determine next/previous focusable items.
+  // We also need to count children with the `active` prop to deal with
+  // instances where `allowMultiple` is set on an accordion with multiple
+  // active children.
+  // We'll memoize these to avoid looping through children unneccessarily.
+  const childrenArray = useMemo(() => {
+    const arr = React.Children.toArray(children);
+    return isArray(arr) ? arr : [];
+  }, [children]);
+
+  const { enabledIndexes, activeChildCount, hasActiveChild } = useMemo(() => {
+    let enabledIndexes = [];
+    let activeChildCount = 0;
+    let hasActiveChild = false;
+    for (let i = 0; i < childrenArray.length; i++) {
+      const child = childrenArray[i];
+      if (!(typeof child.type === "string" || child.props.disabled === true)) {
```

Comment:
What do you think about removing the boolean check for `child.props.disabled` since it is a boolean already?

```jsx
if (!(typeof child.type === "string" || child.props.disabled === true)) {...}

// vs.

if (!(typeof child.type === "string" || child.props.disabled)) {...}
```